controller/stats_reporter.go

```diff
@@ -50,6 +51,31 @@ var (
 	successTagKey    = tag.MustNewKey("success")
 )
 
+type configAwareLatencyMetric struct {
+	delegate kubemetrics.LatencyMetric
+}
+
+func (m configAwareLatencyMetric) Observe(verb string, u url.URL, latency time.Duration) {
+	if !metrics.ShouldEmitMetrics() {
+		return
+	}
+	m.delegate.Observe(verb, u, latency)
+}
+
+type configAwareResultMetric struct {
+	delegate kubemetrics.ResultMetric
+}
+
+func (m configAwareResultMetric) Increment(code string, method string, host string) {
+	if !metrics.ShouldEmitMetrics() {
```

Comment:
Could you explain why calling `Increment` when configmap hasn't been loaded will result in unexpected custom metrics?

Reply:
I assume for the same reason that we don't emit metrics when we Record() when the config isn't loaded.

Reply:
https://github.com/knative/pkg/blob/3c828cf99f67f8066e956490f634ebe90abd6892/metrics/client.go#L33 doesn't seem to call Record.

I can make the method private if I move the configAwareProxy there, though, so I'll do that.